src/drivers/intel/ssp/ssp.c

```diff
@@ -787,6 +788,81 @@ static int ssp_set_config(struct dai *dai, struct ipc_config_dai *common_config,
 	return ret;
 }
 
+static int ssp_set_config_blob(struct dai *dai, struct ipc_config_dai *common_config,
+			       void *spec_config)
+{
+	struct ipc4_ssp_configuration_blob *blob = spec_config;
+	struct ssp_pdata *ssp = dai_get_drvdata(dai);
+	uint32_t ssc0, sstsa, ssrsa;
+
+	/* set config only once for playback or capture */
+	if (dai->sref > 1)
+		return 0;
+
+	ssc0 = blob->i2s_driver_config.i2s_config.ssc0;
+	sstsa = blob->i2s_driver_config.i2s_config.sstsa;
+	ssrsa = blob->i2s_driver_config.i2s_config.ssrsa;
+
+	ssp_write(dai, SSCR0, ssc0);
```

Comment:
We need an incremental change here to mask each register against a bit mask of bits that are allowed to change. i.e. so that no reserved bits are being set.

Reply:
sure, I will add it to my shop list